packages/test/test-utils/src/TestSummaryUtils.ts

```diff
@@ -111,7 +115,23 @@ export async function createSummarizer(
     };
 
     const loader = provider.makeTestLoader(testContainerConfig);
-    return createSummarizerCore(container, loader, summaryVersion);
+    const absoluteUrl = await container.getAbsoluteUrl("");
+    return (await createSummarizerCore(absoluteUrl, loader, summaryVersion)).summarizer;
+}
+
+export async function createSummarizerWithContainer(
+    provider: ITestObjectProvider,
+    absoluteUrl: string | undefined,
+    entryPoint: fluidEntryPoint,
+    loaderProps?: Partial<ILoaderProps>,
+    summaryVersion?: string,
+): Promise<{ container: IContainer; summarizer: ISummarizer; }> {
+    const defaultCodeDetails: IFluidCodeDetails = {
```

Comment:
You should not manually create this. This works because the code details match that of TestObjectProvider. It will break if that changes.

Also, maybe you can export `createSummarizerCore` which returns both the summarizer and the container. This logic can move to the test file. The test-utils should have generic functionality only and this is specific to only one test there.